### ISSUE TITLE
How to handle submission of forms from outside the decorated form component

### ISSUE THREAD

**Issue by JohanNordlinder** [erikras/redux-form#179]
Hello

I'm having difficulties understanding how to handle submission of multiple forms. In my app I have many separate forms and at the bottom of the page I have a *reset, preview, submit* action bar which is not a redux-form component. When I press the *submit* button I want all of the forms to validate (touch all fields) and only if they are all valid the data should be sent to the server.

I've followed the *Responding to Other Actions* topic from the ReadMe and this allows me to listen to the submit event and access the fields for each form in the *pluged in* reducers, but this only allows me to work with one of the forms at a time.

```javascript
export default combineReducers({
  router: routerStateReducer,
  auth,
  form: form.plugin(invoicing()),
});
```
```javascript
export default function invoicing() {
  return {
    invoiceForm: (state, action) => {
      switch (action.type) {
        case 'actionBar/SAVE':
          return {
            ...state,
            // Here I can access the fields from one of the forms at least....
            lastName: {}
          };
        default:
          return state;
      }
    }
  };
}
```

Basically what I (think I) want to do is access the methods touchAll(), valid() and resetForm() from outside the decorated form component. There are of course the exposed action creators which would help me trigger validation of each form but they won't provide me with the valid()-status.

I'm probably missing something here and need some guidance.

Thanks in advance

**Comment by erikras** [comment]
Hmm... and you're certain that it makes sense for them to be separate form components?

I suppose that you could `connect()` your submit button component to the redux-form reducer and manually read all the values, and then you could import your sync validation functions from wherever they are and run them manually to calculate an `isValid` state for all your forms, but that's pretty gross and convoluted. However, nothing is occurring to me for a better way to do it. I would really question your requirement to have them be separate.

...

Oh! How about this? (I'm assuming v2.x)

```js
@connectReduxForm({
  form: 'invoicing',
  fields: ['firstName', 'lastName'],
  validate: validateContactForm
})
class ContactForm extends Component {
  // render contact fields
}

@connectReduxForm({
  form: 'invoicing', // <------ same name
  fields: ['shippingStreet', 'shippingCity'], // <---- different fields
  validate: validateShippingForm
})
class ShippingForm extends Component {
  // render shipping fields
}

@connectReduxForm({
  form: 'invoicing', // <------ same name
  fields: ['billingStreet', 'billingCity'], // <---- different fields
  validate: validateBillingForm
})
class BillingForm extends Component {
  // render billing fields
}

@connectReduxForm({
  form: 'invoicing', // <------ same name
  fields: ['firstName', 'lastName',
    'shippingStreet', 'shippingCity',
    'billingStreet', 'billingCity'], // <---- ALL fields
  validate: somehowCompose(validateContactForm, validateShippingForm, validateBillingForm)
})
class InvoiceSubmitForm extends Component {
  // render submit button
}
```
This way, your submit button component is reading the same data from the Redux state, so the second that the last invalid field is made valid, the submit button could be enabled, because it's listening to those same values, and it can use the regular `handleSubmit` method, which will do the "touch all" for you.

Still kind of gross, but it's the cleanest thing I can come up with if you really can't house all the fields in a single form component.

What do you think?

**Comment by JohanNordlinder** [comment]
Thanks! I implemented this and it worked out fine. Yes I could probably have used a single form in this case but the important bit was being able to separate the submit button from the form component itself as it should be positioned at another place in the DOM/Page, a case which I wouldn't think is so uncommon.

**Comment by erikras** [comment]
Come and get it, folks! [`v4.2.0`](https://github.com/erikras/redux-form/releases/tag/v4.2.0)

**Comment by ben-boxed** [comment]
would be great if you could just reduxForm.submit('myFormName') from an action creator!

**Comment by acontreras89** [comment]
Just in case someone reads this, the `submit('myFormName')` action creator has been out there for some time now, as explained [in the docs](http://redux-form.com/6.8.0/examples/remoteSubmit/).

**Comment by acontreras89** [comment]
The same behavior can be achieved by setting the id to the form element as the form attribute on the submit button:

```html
<form id='myFormId'>
  ...
</form>

<!-- somewhere else -->

<input type='submit' form='myFormId' />
<!-- OR -->
<button type='submit' form='myFormId' />
```

**Comment by gellrich** [comment]
The solution with the form attribute on the submit button don't work in IE. The form attribut is not supported in Internet Explorer. 
There are [polyfills](https://gist.github.com/ChiChou/200362adf9d952abb4fb), but this seemingly don't work with react (?). 
Any ideas, that i can this make work in react? Or any other solutions if i can have the submit button outside the form and have one form (like the [wizard example](https://redux-form.com/7.0.4/examples/wizard/)) but the behavior of a [remote submit](https://redux-form.com/7.0.4/examples/remotesubmit/)?

Many Thanks!

**Comment by Abhijeetjaiswalit** [comment]
@acontreras89 is there any way to get a callback after submit('myFormName') ?

dispatch(submit('myFormName')).then((res) =>{ console.log('res',res) }) is not working. I'm returning something after form submit.
